plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,468 @@
+		return nil
+	}
+
+	first := strings.Replace(line[0], ":", "", 1)
+
+	var eventsFields = []string{
+		"inoderevalidates",
+		"dentryrevalidates",
+		"datainvalidates",
+		"attrinvalidates",
+		"vfsopen",
+		"vfslookup",
+		"vfsaccess",
+		"vfsupdatepage",
+		"vfsreadpage",
+		"vfsreadpages",
+		"vfswritepage",
+		"vfswritepages",
+		"vfsgetdents",
+		"vfssetattr",
+		"vfsflush",
+		"vfsfsync",
+		"vfslock",
+		"vfsrelease",
+		"congestionwait",
+		"setattrtrunc",
+		"extendwrite",
+		"sillyrenames",
+		"shortreads",
+		"shortwrites",
+		"delay",
+		"pnfsreads",
+		"pnfswrites",
+	}
+
+	var bytesFields = []string{
+		"normalreadbytes",
+		"normalwritebytes",
+		"directreadbytes",
+		"directwritebytes",
+		"serverreadbytes",
+		"serverwritebytes",
+		"readpages",
+		"writepages",
+	}
+
+	var xprtudpFields = []string{
+		"bind_count",
+		"rpcsends",
+		"rpcreceives",
+		"badxids",
+		"inflightsends",
+		"backlogutil",
+	}
+
+	var xprttcpFields = []string{
+		"bind_count",
+		"connect_count",
+		"connect_time",
+		"idle_time",
+		"rpcsends",
+		"rpcreceives",
+		"badxids",
+		"inflightsends",
+		"backlogutil",
+	}
+
+	var nfsopFields = []string{
+		"ops",
+		"trans",
+		"timeouts",
+		"bytes_sent",
+		"bytes_recv",
+		"queue_time",
+		"response_time",
+		"total_time",
+		"errors",
+	}
+
+	var fields = make(map[string]interface{})
+
+	switch first {
+	case "READ", "WRITE":
+		fields["ops"] = nline[0]
+		fields["retrans"] = nline[1] - nline[0]
+		fields["bytes"] = nline[3] + nline[4]
+		fields["rtt"] = nline[6]
+		fields["exe"] = nline[7]
+		tags["operation"] = first
+		acc.AddFields("nfsstat", fields, tags)
+	}
+
+	if fullstat {
+		switch first {
+		case "events":
+			if len(nline) >= len(eventsFields) {
+				for i, t := range eventsFields {
+					fields[t] = nline[i]
+				}
+				acc.AddFields("nfs_events", fields, tags)
+			}
+
+		case "bytes":
+			if len(nline) >= len(bytesFields) {
+				for i, t := range bytesFields {
+					fields[t] = nline[i]
+				}
+				acc.AddFields("nfs_bytes", fields, tags)
+			}
+
+		case "xprt":
+			if len(line) > 1 {
+				switch line[1] {
+				case "tcp":
+					if len(nline)+2 >= len(xprttcpFields) {
+						for i, t := range xprttcpFields {
+							fields[t] = nline[i+2]
+						}
+						acc.AddFields("nfs_xprt_tcp", fields, tags)
+					}
+				case "udp":
+					if len(nline)+2 >= len(xprtudpFields) {
+						for i, t := range xprtudpFields {
+							fields[t] = nline[i+2]
+						}
+						acc.AddFields("nfs_xprt_udp", fields, tags)
+					}
+				}
+			}
+		}
+
+		switch version {
+		case "3":
+			tags["operation"] = first
+			if nfs3Ops[first] && (len(nline) <= len(nfsopFields)) {
+				for i, t := range nline {
+					fields[nfsopFields[i]] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		case "4":
+			tags["operation"] = first
+			if nfs4Ops[first] && (len(nline) <= len(nfsopFields)) {
+				for i, t := range nline {
+					fields[nfsopFields[i]] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		}
```

Comment:
Can we squash this to
```suggestion
		if (version == "3" && nfs3Ops[first]) || (version == "4" && nfs4Ops[first]) {
			tags["operation"] = first
			if len(nline) <= len(nfsopFields) {
				for i, t := range nline {
					fields[nfsopFields[i]] = t
				}
				acc.AddFields("nfs_ops", fields, tags)
			}
		}
```
please!?